traversal/procedure/ProcedureEdge.java

```diff
@@ -1186,16 +1178,21 @@ public boolean isClosure(GraphManager graphMgr, Vertex<?, ?> fromVertex, Vertex<
                                                 .edge(ROLEPLAYER, rt, player.iid().prefix(), player.iid().type())
                                                 .toAndOptimised(),
                                         ASC
-                                ).filter(kv -> kv.key().equals(player) && !scoped.contains(kv.value()));
+                                ).filter(kv -> kv.key().equals(player));
```

Comment:
TODO: try to re-instante this

Reply:
So I attemped to revert back to using the role scope filtering during generation, rather than at vertex verification time.

The main issue is that we are no longer moving forward and backtracking based on edges, but instead we are doing vertex-by-vertex.

So now imagine a plan:
```
0: $r [friendship]
1: $x [dog, person]
  $r --RolePlayer--> $x
2: $y [dog]
  $r --RolePlayer--> $y
3: $a [dog-name]
  $y --Has--> $a
```

In this case, traversing `$r` will forward verify the edges generating `$y` and `$x`. However, `$x` is checked first and occupies the only role leading to a `dog`. We then verify that `$y` has an instance. In the old "filtering" model, we will immediately filter out the only role leading to a valid `dog`, since that is occupied by the traversal for `$x`. However, since we don't want to have multiple types of backtracking, we can't re-visit `$x` to get the next person and free up the `dog` role for `$y` to use.

Even if we did implement edge backtracking during the edge lookahead, the cases for backtracking later are difficult. Imagine the first intersections succeed, and we find a `$x` and `$y`. However, the _next_ `$y` may not be found because of `$x`, so we need to know that `$y` could have failed because of `$x` - tracking this causality with the filtering becomes really hard.